### DISCUSSION TITLE
@ConfigProperty annotation

### DISCUSSION THREAD

**Discussion by Snorri1986** [quarkusio/quarkus#31870] [Community]
Why `@ConfigProperty` annotation does not initialize variables in a class ?
Instead, I use `ConfigProvider.getConfig().getValue("system.varname", String.class)`
The article below tells me that it should work.
https://quarkus.io/guides/config 
Kindly ask you to explain in what cases better use `@ConfigProperty` and in what cases `ConfigProvider` ?
Thanks.

**Comment by radcortez** [comment]
The `@ConfigProperty` annotation only works for CDI-based beans (managed by Arc): https://quarkus.io/guides/cdi-reference. 

The programmatic API can be used in cases where you need to look up a configuration in a non-CDI environment or when the piece of code is not a CDI Bean.
